lib/simpler/router/route.rb

```diff
@@ -12,9 +12,30 @@ def initialize(method, path, controller, action)
       end
 
       def match?(method, path)
-        @method == method && path.match(@path)
+        @method == method && path_match?(path)
       end
 
+      def variables(path)
+        path_request = split_path(path)
+        path_route = split_path(@path)
+        variables = {}
+        path_route.zip(path_request).each { |route, request| variables[route[1..-1].to_sym] = request if route[0] == ':' }
```

Comment:
Когда тебе из коллекции надо получить другую коллекцию - используй `each.with_object`

Reply:
Вот так можно без мутации внешней переменной:
```suggestion
        path_route.zip(path_request).each.with_object({}) do |(route, request), variables| 
          variables[route[1..-1].to_sym] = request if route[0] == ':' 
        end
```